src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetadataResponse.cs

```diff
@@ -94,10 +94,7 @@ internal string GetValueForField(string fieldName)
                 break;
         }
 
-        if (amsValue == null)
-        {
-            amsValue = string.Empty;
-        }
+        amsValue ??= string.Empty;
```

Comment:
nit: Use `string amsValue = string.Empty;` at line 58 to get rid of this.

Reply:
It can get overwritten by one of the cases so still need this check.

Reply:
Ahh, these values that we are assigning could be `null`.